File: Service/ClientCare/RMA.Service.ClientCare.Contracts/Entities/Member/Declaration.cs
using RMA.Service.Admin.MasterDataManager.Contracts.Enums;

using System.Collections.Generic;

namespace RMA.Service.ClientCare.Contracts.Entities.Member
{
    public class Declaration
    {
        public int DeclarationId { get; set; } // DeclarationId (Primary key)
        public DeclarationStatusEnum DeclarationStatus { get; set; } // DeclarationStatusId
        public DeclarationTypeEnum DeclarationType { get; set; } // DeclarationTypeId
        public DeclarationRenewalStatusEnum DeclarationRenewalStatus { get; set; } // DeclarationRenewalStatusId
        public int RolePlayerId { get; set; } // RolePlayerId
        public int DeclarationYear { get; set; } // DeclarationYear
        public int? ProductOptionId { get; set; } // ProductOptionId
        public int? AverageEmployeeCount { get; set; } // AverageEmployeeCount
        public decimal? AverageEarnings { get; set; } // AverageEarnings
        public decimal? PenaltyPercentage { get; set; } // PenaltyPercentage
        public decimal? Rate { get; set; } // Rate
        public decimal? Premium { get; set; } // Premium
        public decimal? PenaltyRate { get; set; } // PenaltyRate
        public decimal? PenaltyPremium { get; set; } // PenaltyPremium
        public decimal? Adjustment { get; set; } // Adjustment
        public string Comment { get; set; } // Comment (length: 255)
        public bool IsDeleted { get; set; } // IsDeleted
        public string CreatedBy { get; set; } // CreatedBy (length: 50)
        public System.DateTime CreatedDate { get; set; } // CreatedDate
        public string ModifiedBy { get; set; } // ModifiedBy (length: 50)
        public System.DateTime ModifiedDate { get; set; } // ModifiedDate

        public List<DeclarationAllowance> DeclarationAllowances { get; set; }
        public List<Declaration> DependentDeclarations { get; set; }
        public List<DeclarationBillingIntegration> DeclarationBillingIntegrations { get; set; }
    }
}

File: Web/RMA-Member-Portal/memberportal-app/src/app/shared/models/declaration.ts
import { DeclarationRenewalStatusEnum } from "../enums/declaration-renewal-status.enum";
import { DeclarationStatusEnum } from "../enums/declaration-status.enum";
import { DeclarationTypeEnum } from "../enums/declaration-type.enum";
import { DeclarationAllowance } from "./declaration-allowance";
import { DeclarationBillingIntegration } from "./declaration-billing-integration";


export class Declaration {
  declarationId: number;
  declarationStatus: DeclarationStatusEnum;
  declarationType: DeclarationTypeEnum;
  rolePlayerId: number;
  averageEmployeeCount: number;
  averageEarnings: number;
  declarationYear: number;
  declarationRenewalStatus: DeclarationRenewalStatusEnum;
  comment: string;
  rate: number;
  premium: number;
  penaltyPercentage: number;
  penaltyRate: number;
  penaltyPremium: number;
  adjustment: number;
  productOptionId: number;
  declarationAllowances: DeclarationAllowance[] = [];
  variancePercentage: number;
  dependentDeclarations: Declaration[];
  declarationBillingIntegrations: DeclarationBillingIntegration[];

  isDeleted: boolean;
  createdBy: string;
  createdDate: Date;
  modifiedBy: string;
  modifiedDate: Date;

}

